tests/integration/sugar/test_transaction.py

```diff
@@ -354,3 +356,122 @@ async def test_reliable_submission_no_last_ledger_sequence(self, client):
         signed_payment_transaction = await sign(payment_transaction, WALLET)
         with self.assertRaises(XRPLReliableSubmissionException):
             await send_reliable_submission(signed_payment_transaction, client)
+
+
+class TestSubmitAndWait(IntegrationTestCase):
+    @test_async_and_sync(
+        globals(),
+        [
+            "xrpl.transaction.submit_and_wait",
+            "xrpl.account.get_next_valid_seq_number",
+            "xrpl.ledger.get_fee",
+        ],
+    )
+    async def test_submit_and_wait_simple(self, client):
+        account_set = AccountSet(
+            account=ACCOUNT,
+            set_flag=SET_FLAG,
+        )
+        await accept_ledger_async(delay=1)
+        response = await submit_and_wait(account_set, client, WALLET)
+        self.assertTrue(response.result["validated"])
+        self.assertEqual(response.result["meta"]["TransactionResult"], "tesSUCCESS")
+        self.assertTrue(response.is_successful())
+        self.assertEqual(response.result["Fee"], await get_fee(client))
+        WALLET.sequence += 1
+
+    @test_async_and_sync(
+        globals(),
+        [
+            "xrpl.transaction.submit_and_wait",
+            "xrpl.account.get_next_valid_seq_number",
+            "xrpl.ledger.get_fee",
+        ],
+    )
+    async def test_submit_and_wait_payment(self, client):
+        payment_transaction = Payment(
+            account=ACCOUNT,
+            amount="10",
+            destination=DESTINATION,
+        )
+        await accept_ledger_async(delay=1)
+        response = await submit_and_wait(payment_transaction, client, WALLET)
+        self.assertTrue(response.result["validated"])
+        self.assertEqual(response.result["meta"]["TransactionResult"], "tesSUCCESS")
+        self.assertTrue(response.is_successful())
+        self.assertEqual(response.result["Fee"], await get_fee(client))
+        WALLET.sequence += 1
+
```

Comment:
There should probably be a test for "fail hard" as well if that's a new feature :)

Reply:
I'm not sure how to test fail hard since its logic is implemented on rippled server. The client library only made a request to the server with a fail hard variable that has already been defined here:

https://github.com/XRPLF/xrpl-py/blob/a6469f131ea2ccc964b43d394d1ff23145cc2da5/xrpl/models/requests/submit_only.py#L32-L63

Reply:
It's not easily testable, it's not worth testing IMO.